tests_integration/conftest.py

```diff
@@ -100,6 +104,15 @@ def use_config(self, config_file_name: str) -> None:
     def invoke(self, *a, **kw):
         if "catch_exceptions" not in kw:
             kw.update(catch_exceptions=False)
+
+        # The CLI automatically appends $USER to some created resource identifiers,
+        # let's use this behaviour to add a unique suffix to resources
+        # created/updated/dropped in tests to allow us to run simultaneous instances
+        env = kw.get("env", {}).copy()
+        env["USER"] = (
+            f"{env['USER']}_{self._user_suffix}" if "USER" in env else self._user_suffix
```

Comment:
outside this class, it's called `identifier_suffix` since it's meant to be a generic concept. inside the class it's called `user_suffix` to hint at how it's used.

I still need to append if USER is passed in since it's adding a random ID and the test name (tests no longer pass in the random ID, and if they pass in a static value for USER, we still need to append the random ID to avoid concurrecy conflicts)

Reply:
updated: [`d0c7a32` (#1437)](https://github.com/snowflakedb/snowflake-cli/pull/1437/commits/d0c7a32bf8c000ab1693838286cc2d38cfad47fb)

Reply:
It's confusing that somehow USER becomes identifier_suffix magically. If you pass in a USER, then USER is also mutated to a different value.
 
1. I understand that you want to make this reusable. In that case, do we need _suffix? it's more like a test_identifier.

2. If I pass in USER in env variables in a test, I do not expect it to be mutated. My test could be focused on testing USER specific logic, and I should be able to do that safely.

Reply:
Based on our offline discussion, this is ok for now until we have the long term solution.